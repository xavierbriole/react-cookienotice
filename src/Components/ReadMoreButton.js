// @flow

import * as React from 'react'
import styled from 'styled-components'

const Button = styled.a`
  font-size: 16px;
  height: auto;
  color: #b5b5b5;
  padding: 10px 18px;
  border: 1px solid #b5b5b5;
  border-radius: 24px;
  box-shadow: none;
  text-decoration: none;
  white-space: nowrap;
  transition: all 0.5s ease 0s;

  &:hover {
    color: #000000;
    border-color: #000000;
  }

  &:not(:first-child) {
    margin-left: 8px;
  }

  @media (max-width: 768px) {
    &:not(:first-child) {
      margin-left: 0;
    }

    font-size: 12px;
    margin: 5px 0;
  }

  @media (prefers-color-scheme: dark) {
    &:hover {
      color: #ffffff;
      border-color: #ffffff;
    }
  }
`

type Props = {|
  label: string,
  link: string,
  openInNewTab: boolean,
|}

export default class ReadMoreButton extends React.Component<Props> {
  render() {
    const { label, link, openInNewTab } = this.props

    return (
      <Button
        className='read-more-button'
        href={link}
        target={openInNewTab ? '_blank' : '_self'}
      >
        {label}
      </Button>
    )
  }
}
