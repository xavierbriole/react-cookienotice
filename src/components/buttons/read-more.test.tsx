import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import ReadMoreButton from './read-more'

describe('ReadMoreButton', () => {
  describe('with open in new tab false', () => {
    it('should render', () => {
      const { container } = render(
        <ReadMoreButton
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={false}
        />,
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it('should have right attributes', () => {
      const { getByText } = render(
        <ReadMoreButton
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={false}
        />,
      )

      const readMoreButton = getByText('readMoreButtonLabel')

      expect(readMoreButton.getAttribute('target')).toBe('_self')
      expect(readMoreButton.getAttribute('rel')).toBe(null)
    })
  })

  describe('with open in new tab true', () => {
    it('should render', () => {
      const { container } = render(
        <ReadMoreButton
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={true}
        />,
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it('should have right attributes', () => {
      const { getByText } = render(
        <ReadMoreButton
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={true}
        />,
      )

      const readMoreButton = getByText('readMoreButtonLabel')

      expect(readMoreButton.getAttribute('target')).toBe('_blank')
      expect(readMoreButton.getAttribute('rel')).toBe('noreferrer')
    })
  })

  it('should display read more button label', () => {
    const { getByText } = render(
      <ReadMoreButton
        readMoreButtonLabel='readMoreButtonLabel'
        readMoreButtonLink='readMoreButtonLink'
        readMoreButtonOpenInNewTab={false}
      />,
    )

    expect(getByText('readMoreButtonLabel')).toBeInTheDocument()
  })
})
