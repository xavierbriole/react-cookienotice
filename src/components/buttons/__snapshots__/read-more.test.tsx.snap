// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMoreButton with open in new tab false should render 1`] = `
<a
  class="react-cookienotice-read-more-button"
  href="readMoreButtonLink"
  target="_self"
>
  readMoreButtonLabel
</a>
`;

exports[`ReadMoreButton with open in new tab true should render 1`] = `
<a
  class="react-cookienotice-read-more-button"
  href="readMoreButtonLink"
  rel="noreferrer"
  target="_blank"
>
  readMoreButtonLabel
</a>
`;
